extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/ForwardedParser.java

```diff
@@ -137,6 +154,11 @@ private void calculate() {
             String hostHeader = delegate.getHeader(X_FORWARDED_HOST);
             if (hostHeader != null) {
                 setHostAndPort(hostHeader.split(",")[0], port);
+            } else {
```

Comment:
Hi @ejba , with an apache proxy this won't do the job. As documented in https://httpd.apache.org/docs/2.4/mod/mod_proxy.html both headers can be there, but X_FORWARDED_SERVER is the correct one. Is it a big effort to trigger that by a property, e.g. `quarkus.http.proxy.forward_header=x_forwarded_host/x_forwarded_server/referer` ?

Reply:
I could also give preference to X-Forwarded-Server instead of X-Forwarded-Host. 
@stuartwdouglas @sberyozkin what do you prefer?

Reply:
I guess for backward-compatibility is would be good to keep the previous logic as default, but give the user the possibilty to change the header in use.
For an apache proxy it must be the X_FORWARDED_SERVER. I don't know the behaviour of other http servers, but there must be a reason why @stuartwdouglas implemented it like this.
Configuring that by a new property is probably more work, but would be a more stable solution.

The third option with the referer is just another idea without actual current need, but maybe could give more options to future users with yet another proxy. In that case the host has to be extracted from the referer uri. That sould be cnsidered for the discussion.

Reply:
@ejba, @kraeftbraeu, the filter is enabled [here](https://github.com/quarkusio/quarkus/blob/ac1cdf32b3af0eb3c198f9028f4512326b3fefba/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java#L298).
So I suppose we can have `httpConfiguration.forwardedHostHeader` enum property introduced (default is `X-FORWARDED-HOST` to preserve the backward compatibility, and X_FORWARDED_SERVER), and pass it to the filter to use. 
How does it sound ?
CC @stuartwdouglas

Reply:
:THUMBSUP from my side